src/core/algorithms/dc/verifier/dc_verifier.cpp

```diff
@@ -0,0 +1,523 @@
+            res.push_back(pred);
+        }
+    }
+
+    return res;
+}
+
+std::vector<DC> DCVerifier::ConvertDisequalities(const DC& dc) {
+    std::vector<DC> res;
+    std::vector<Predicate> no_diseq_preds, diseq_preds, predicates = dc.GetPredicates();
+
+    for (auto& pred : predicates) {
+        auto op = pred.GetOperator();
+        if (op.GetType() == OperatorType::kUnequal and pred.IsCrossTuple())
+            diseq_preds.push_back(std::move(pred));
+        else
+            no_diseq_preds.push_back(std::move(pred));
+    }
+
+    std::vector<std::vector<Predicate>> temp_preds{std::move(no_diseq_preds)};
+    for (auto const& pred : diseq_preds) {
+        size_t size = temp_preds.size();
+        for (size_t i = 0; i < size; i++) {
+            temp_preds.push_back(temp_preds[i]);
+        }
+
+        ColumnOperand left = pred.GetLeftOperand();
+        ColumnOperand right = pred.GetRightOperand();
+        auto less_pred = Predicate(Operator(OperatorType::kLess), left, right);
+        auto greater_pred = Predicate(Operator(OperatorType::kGreater), left, right);
+
+        for (size_t i = 0; i < temp_preds.size() / 2; i++) {
+            temp_preds[i].push_back(less_pred);
+        }
+
+        for (size_t i = temp_preds.size() / 2; i < temp_preds.size(); i++) {
+            temp_preds[i].push_back(greater_pred);
+        }
+    }
+
+    for (auto& vec_pred : temp_preds) {
+        res.push_back({std::make_move_iterator(vec_pred.begin()),
+                       std::make_move_iterator(vec_pred.end())});
+    }
+
+    return res;
+}
+
+// FIX: DC may contain braces, numbers or spaces, naive separation doesn't belong here.
+std::vector<Predicate> DCVerifier::SplitDC(std::string const& dc_string) {
+    size_t ind, start = 0;
+    std::string token, sep = " and ";
+    std::vector<Predicate> predicates;
+    while (true) {
+        ind = dc_string.find(sep, start);
+        token = dc_string.substr(start, ind - start);
+        std::vector<std::string> predicate_parts;
+        boost::split(predicate_parts, token, boost::is_any_of(" "));
+        for (size_t i = 0; i < 3; i += 2) {
+            std::replace_if(predicate_parts[i].begin(), predicate_parts[i].end(),
+                            boost::is_any_of("!()"), ' ');
+            boost::trim(predicate_parts[i]);
+        }
+        auto left_op = ColumnOperand(predicate_parts[0], *relation_->GetSchema());
+        auto right_op = ColumnOperand(predicate_parts[2], *relation_->GetSchema());
+        auto oper = Operator::kStringToOperatorType.at(frozen::string(predicate_parts[1]));
+        predicates.push_back({oper, left_op, right_op});
+
+        if (ind == std::string::npos) break;
+        start = ind + sep.size();
+    }
+
+    return predicates;
+}
+
+std::vector<std::byte const*> DCVerifier::GetTuple(size_t row) {
+    std::vector<std::byte const*> res;
+    res.reserve(data_.size());
+    for (auto const& col : data_) {
+        res.push_back(col.GetValue(row));
+    }
```

Comment:
```suggestion
    std::vector<std::byte const*> res;
    std::transform(data_.begin(), data_.end(), std::back_inserter(res), [&row](model::TypedColumnData& col) { return col.GetValue(row); });
```
or, if you want to pre-allocate memory,
```suggestion
    std::vector<std::byte const*> res(data_.size());
    std::transform(data_.begin(), data_.end(), res.begin(), [&row](model::TypedColumnData& col) { return col.GetValue(row); });
```